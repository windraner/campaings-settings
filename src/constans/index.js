export const MENU_LIST = [
  'Campaign details',
  'Media Collections',
  'Blacklists',
  'Media rights'
];

export const ACTIVE_TAB = 'activeTab';
export const NAME = 'name';
export const DESCRIPTION = 'description';
export const PRODUCT_CATALOGUE = 'productCatalogue';
export const IS_SHOPPABLE_LINK = 'isShoppableLink';
export const IS_MEDIA_COLLECTION = 'isMediaCollection';
export const MEDIA_COLLECTION_LIST = 'mediaCollectionList';
export const IS_PHOTO_MODERATION_QUEUE = 'isPhotosModerationQueue';
export const USER_BLACK_LIST = 'userBlackList';
export const HASHTAG_BLACK_LIST = 'hashtagBlackList';
export const ACCOUNTS_SETTINGS_ITEMS = 'accountsSettingsItems';
export const SHOW_ERROR_ALERT = 'showErrorAlert';
export const SHOW_SUCCESS_ALERT = 'showSuccessAlert';
export const IS_MEDIA_RIGHTS = 'isMediaRights';
export const HASHTAG_RIGHTS_LIST = 'hashtagRightsList';
export const TERMS_AND_CONDITIONS = 'termsAndConditions';

export const VALIDATED_FIELDS = [
  ACTIVE_TAB
];
