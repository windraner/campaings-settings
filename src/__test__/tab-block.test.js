import React from 'react';
import TabBlock from '../components/main-content-section/tab-block/tab-block';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import image from '../test.png';
import toJson from 'enzyme-to-json';

import { MENU_LIST } from '../constans';

configure({ adapter: new Adapter() });

const mock = {
  userBlackList: ['asd', 'fgh'],
  description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit',
  hashtagBlackList: ['zxc', 'vbn'],
  isMediaCollection: true,
  isPhotosModerationQueue: true,
  isShoppableLink: true,
  mediaCollectionList: ['qwe', 'rty'],
  name: 'Summer Sales 10%',
  productCatalogue: 'Product catalogue2',
  productsCatalogueList: [
    'Product catalogue1',
    'Product catalogue2',
    'Product catalogue3',
    'Product catalogue4',
  ],
  stateHandler() {},
  switchHandler() {},
  accountsSettingsItems: [
    {
      id: '1',
      url: image,
      userName : 'Instagram',
      platform: 'Instagram',
      status: 'Verified',
      userFeed: true,
      taggedFeed: true,
      mentionedFeed: false
    },
    {
      id: '2',
      url: image,
      userName : 'Blackmirror',
      platform: 'Twitter',
      status: 'Rejected',
      userFeed: false,
      taggedFeed: false,
      mentionedFeed: false
    }
  ],
};

describe('<TabBlock />', () => {
  it('matches the snapshot tab 1', () => {
    const tree = shallow(<TabBlock {...mock} activeTab={MENU_LIST[0]} />);
    expect(toJson(tree)).toMatchSnapshot();
  });
  it('matches the snapshot tab 2', () => {
    const tree = shallow(<TabBlock {...mock} activeTab={MENU_LIST[1]} />);
    expect(toJson(tree)).toMatchSnapshot();
  });
  it('matches the snapshot tab 3', () => {
    const tree = shallow(<TabBlock {...mock} activeTab={MENU_LIST[2]} />);
    expect(toJson(tree)).toMatchSnapshot();
  });
  it('matches the snapshot tab 4', () => {
    const tree = shallow(<TabBlock {...mock} activeTab={MENU_LIST[3]} />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
