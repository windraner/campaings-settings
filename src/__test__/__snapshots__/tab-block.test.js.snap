// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TabBlock /> matches the snapshot tab 1 1`] = `
<div
  className="campaigns-settings__tab-block-container"
>
  <div
    className="campaigns-settings__tab-block-wrapper"
  >
    <Tab1
      accountsSettingsItems={
        Array [
          Object {
            "id": "1",
            "mentionedFeed": false,
            "platform": "Instagram",
            "status": "Verified",
            "taggedFeed": true,
            "url": "test.png",
            "userFeed": true,
            "userName": "Instagram",
          },
          Object {
            "id": "2",
            "mentionedFeed": false,
            "platform": "Twitter",
            "status": "Rejected",
            "taggedFeed": false,
            "url": "test.png",
            "userFeed": false,
            "userName": "Blackmirror",
          },
        ]
      }
      activeTab="Campaign details"
      description="Lorem ipsum dolor sit amet, consectetur adipiscing elit"
      hashtagBlackList={
        Array [
          "zxc",
          "vbn",
        ]
      }
      isMediaCollection={true}
      isPhotosModerationQueue={true}
      isShoppableLink={true}
      mediaCollectionList={
        Array [
          "qwe",
          "rty",
        ]
      }
      name="Summer Sales 10%"
      productCatalogue="Product catalogue2"
      productsCatalogueList={
        Array [
          "Product catalogue1",
          "Product catalogue2",
          "Product catalogue3",
          "Product catalogue4",
        ]
      }
      stateHandler={[Function]}
      switchHandler={[Function]}
      userBlackList={
        Array [
          "asd",
          "fgh",
        ]
      }
    />
  </div>
</div>
`;

exports[`<TabBlock /> matches the snapshot tab 2 1`] = `
<div
  className="campaigns-settings__tab-block-container"
>
  <div
    className="campaigns-settings__tab-block-wrapper"
  >
    <Tab2
      accountsSettingsItems={
        Array [
          Object {
            "id": "1",
            "mentionedFeed": false,
            "platform": "Instagram",
            "status": "Verified",
            "taggedFeed": true,
            "url": "test.png",
            "userFeed": true,
            "userName": "Instagram",
          },
          Object {
            "id": "2",
            "mentionedFeed": false,
            "platform": "Twitter",
            "status": "Rejected",
            "taggedFeed": false,
            "url": "test.png",
            "userFeed": false,
            "userName": "Blackmirror",
          },
        ]
      }
      activeTab="Media Collections"
      description="Lorem ipsum dolor sit amet, consectetur adipiscing elit"
      hashtagBlackList={
        Array [
          "zxc",
          "vbn",
        ]
      }
      isMediaCollection={true}
      isPhotosModerationQueue={true}
      isShoppableLink={true}
      mediaCollectionList={
        Array [
          "qwe",
          "rty",
        ]
      }
      name="Summer Sales 10%"
      productCatalogue="Product catalogue2"
      productsCatalogueList={
        Array [
          "Product catalogue1",
          "Product catalogue2",
          "Product catalogue3",
          "Product catalogue4",
        ]
      }
      stateHandler={[Function]}
      switchHandler={[Function]}
      userBlackList={
        Array [
          "asd",
          "fgh",
        ]
      }
    />
  </div>
</div>
`;

exports[`<TabBlock /> matches the snapshot tab 3 1`] = `
<div
  className="campaigns-settings__tab-block-container"
>
  <div
    className="campaigns-settings__tab-block-wrapper"
  >
    <Tab3
      accountsSettingsItems={
        Array [
          Object {
            "id": "1",
            "mentionedFeed": false,
            "platform": "Instagram",
            "status": "Verified",
            "taggedFeed": true,
            "url": "test.png",
            "userFeed": true,
            "userName": "Instagram",
          },
          Object {
            "id": "2",
            "mentionedFeed": false,
            "platform": "Twitter",
            "status": "Rejected",
            "taggedFeed": false,
            "url": "test.png",
            "userFeed": false,
            "userName": "Blackmirror",
          },
        ]
      }
      activeTab="Blacklists"
      description="Lorem ipsum dolor sit amet, consectetur adipiscing elit"
      hashtagBlackList={
        Array [
          "zxc",
          "vbn",
        ]
      }
      isMediaCollection={true}
      isPhotosModerationQueue={true}
      isShoppableLink={true}
      mediaCollectionList={
        Array [
          "qwe",
          "rty",
        ]
      }
      name="Summer Sales 10%"
      productCatalogue="Product catalogue2"
      productsCatalogueList={
        Array [
          "Product catalogue1",
          "Product catalogue2",
          "Product catalogue3",
          "Product catalogue4",
        ]
      }
      stateHandler={[Function]}
      switchHandler={[Function]}
      userBlackList={
        Array [
          "asd",
          "fgh",
        ]
      }
    />
  </div>
</div>
`;

exports[`<TabBlock /> matches the snapshot tab 4 1`] = `
<div
  className="campaigns-settings__tab-block-container"
>
  <div
    className="campaigns-settings__tab-block-wrapper"
  >
    <Tab4
      accountsSettingsItems={
        Array [
          Object {
            "id": "1",
            "mentionedFeed": false,
            "platform": "Instagram",
            "status": "Verified",
            "taggedFeed": true,
            "url": "test.png",
            "userFeed": true,
            "userName": "Instagram",
          },
          Object {
            "id": "2",
            "mentionedFeed": false,
            "platform": "Twitter",
            "status": "Rejected",
            "taggedFeed": false,
            "url": "test.png",
            "userFeed": false,
            "userName": "Blackmirror",
          },
        ]
      }
      activeTab="Media rights"
      description="Lorem ipsum dolor sit amet, consectetur adipiscing elit"
      hashtagBlackList={
        Array [
          "zxc",
          "vbn",
        ]
      }
      isMediaCollection={true}
      isPhotosModerationQueue={true}
      isShoppableLink={true}
      mediaCollectionList={
        Array [
          "qwe",
          "rty",
        ]
      }
      name="Summer Sales 10%"
      productCatalogue="Product catalogue2"
      productsCatalogueList={
        Array [
          "Product catalogue1",
          "Product catalogue2",
          "Product catalogue3",
          "Product catalogue4",
        ]
      }
      stateHandler={[Function]}
      switchHandler={[Function]}
      userBlackList={
        Array [
          "asd",
          "fgh",
        ]
      }
    />
  </div>
</div>
`;
